### PR TITLE
Consolidate natures

### PR DESCRIPTION
**Pull request by fdeblasio**
## Description
Consolidates various nature values into a new NatureInfo struct and refactors the nature stat table

## **Discord contact info**
Frankfurter0

### CODE REVIEW

**Comment by Bassoonian** [comment]
Would it be possible to refactor `gNatureStatTable` so this new consolidated nature struct would include which stat it raises and which stat it lowers, if any? I know it's a refactor, so feel free to indicate if you're not comfortable with it, but it's the only bit of natures that's still separated :)

File: src/pokemon.c [hunk 1]
```diff
@@ -358,62 +358,293 @@ const struct SpindaSpot gSpindaSpotGraphics[] =
...
+// Support percentages are listed in comments off to the side instead
+#define PALACE_STYLE(atk, def, atkLow, defLow) {atk, atk + def, atkLow, atkLow + defLow}
+
+// The below data for smokescreen starts and ends with some data that belongs to battle_gfx_sfx_util.c
+
+const struct NatureInfo gNaturesInfo[NUM_NATURES] =
+{
+    [NATURE_HARDY] = {
+        .name = COMPOUND_STRING("Hardy"),
+        .statUp = STAT_HP,
```
Comment (Bassoonian) [inline src/pokemon.c:375]: All neutral natures have a particular stat for ups and downs that affects their preferred/disliked flavour. [see Bulbapedia](https://bulbapedia.bulbagarden.net/wiki/Nature#List_of_Natures)

File: src/pokemon_summary_screen.c [hunk 2]
```diff
@@ -3481,19 +3481,21 @@ static void PrintRibbonCount(void)
     PrintTextOnWindow(AddWindowFromTemplateList(sPageSkillsTemplate, PSS_DATA_WINDOW_SKILLS_RIBBON_COUNT), text, x, 1, 0, 0);
 }
 
-static void BufferStat(u8 *dst, s8 natureMod, u32 stat, u32 strId, u32 n)
+static void BufferStat(u8 *dst, u8 statIndex, u32 stat, u32 strId, u32 n)
```
Comment (Bassoonian) [inline src/pokemon_summary_screen.c:3484]: This bit will need to be rewritten so it uses the neutral string if the up and down stats are equal (see comment above)

File: src/battle_dome.c [hunk 3]
```diff
@@ -4410,11 +4410,11 @@ static void DisplayTrainerInfoOnCard(u8 flags, u8 trainerTourneyId)
                 else
                     nature = gSaveBlock2Ptr->frontier.domePlayerPartyData[i].nature;
 
-                if (gNatureStatTable[nature][j] > 0)
+                if (gNaturesInfo[nature].statUp == NUM_STATS + 1)
```
Comment (Bassoonian) [inline src/battle_dome.c:4413]: I'm not sure how this works without taking j into account. Are you sure this is how you intended it? Same for the three similar occurrences below

**Review by Bassoonian — CHANGES_REQUESTED** [review]

File: src/pokemon.c [hunk 4]
```diff
@@ -358,62 +358,293 @@ const struct SpindaSpot gSpindaSpotGraphics[] =
...
+// and a separate percent chance for each group when at or below 50% HP
+// The table below doesn't list percentages for Support because you can subtract the other two
+// Support percentages are listed in comments off to the side instead
+#define PALACE_STYLE(atk, def, atkLow, defLow) {atk, atk + def, atkLow, atkLow + defLow}
+
+// The below data for smokescreen starts and ends with some data that belongs to battle_gfx_sfx_util.c
+
+const struct NatureInfo gNaturesInfo[NUM_NATURES] =
+{
+    [NATURE_HARDY] = {
```
Comment (Bassoonian) [inline src/pokemon.c:373]: The braces need to start on a newline. This should be the only change that still needs to happen :)

Comment (fdeblasio) [reply src/pokemon.c]: Fixed!

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**